File: Array/Python/Longest Consecutive Sequence.py
# Problem link: https://leetcode.com/problems/longest-consecutive-sequence/description/
# Time: O(n)
# Space: O(n)
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numsSet  = set(nums)
        result = size = 0

        for i in nums:
            if i - 1 not in numsSet:
                size = 0

                while i + size in numsSet:
                    size += 1
                
                result = max(result, size)
            
        return result

File: Array/Java/Longest Consecutive Sequence.java
// Problem link: https://leetcode.com/problems/longest-consecutive-sequence/description/
// Time: O(n)
// Space: O(n)
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> numsSet = new HashSet<>();

        for(int n : nums) {
            numsSet.add(n);
        }
        
        int result = 0, size;

        for(int i = 0; i < nums.length; i++) {
            
            if(!numsSet.contains(nums[i] - 1)) {
                size = 0;

                while(numsSet.contains(nums[i] + size)) {
                    size += 1;
                }

                result = Math.max(size, result);
            }
        }
        return result;
    }
}
